services/meta/store_fsm.go

```diff
@@ -540,6 +542,23 @@ func (fsm *storeFSM) applyCreateDataNodeCommand(cmd *internal.Command) interface
 	return nil
 }
 
+func (fsm *storeFSM) applyDeleteDataNodeCommand(cmd *internal.Command) interface{} {
+	ext, _ := proto.GetExtension(cmd, internal.E_DeleteDataNodeCommand_Command)
+	v := ext.(*internal.DeleteDataNodeCommand)
+
+	other := fsm.data.Clone()
+	node := other.DataNode(v.GetID())
+	if node == nil {
+		return ErrNodeNotFound
```

Comment:
meh, I don't think this is one of those things that needs to be idempotent. Usually a user triggering it so if they mess up the ID they'd probably like to know.